### PR TITLE
Add custom MLIR-TensorRT usage instruction to Tripy CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jhalakpatel**

### CODE REVIEW

File: tripy/CONTRIBUTING.md [hunk 1]
```diff
@@ -105,12 +105,43 @@ Please make sure any contributions you make satisfy the developer certificate of
...
+1. [Build custom MLIR-TensorRT](#contributing-to-mlir-tensorrt)
+
+2. Launch the container; from the [`tripy` root directory](.) with mlir-tensorrt repository mapped for accessing wheels files, run:
+    ```bash
+    docker run --gpus all -it -v $(pwd):/tripy/ -v $(pwd)/../mlir-tensorrt:/mlir-tensorrt  --rm tripy:latest
+    ```
+
+3. Install MLIR-TensorRT wheels
+    MLIR-TensorRT builds with a specific version of TensorRT, ensure it matches the version required by Tripy.
+    This requirement can be relaxed in future.
```
Comment (jhalakpatel) [inline tripy/CONTRIBUTING.md:122]: @parthchadha Looks like we build MLIR-TensorRT container using [mlir-tensorrt-compiler==0.1.29+cuda12.trt102](https://github.com/NVIDIA/TensorRT-Incubator/blob/main/tripy/pyproject.toml#L10) and install `TensorRT-10.1.0.27` in the [Dockerfile](https://github.com/NVIDIA/TensorRT-Incubator/blob/main/tripy/Dockerfile#L28).

Comment (parthchadha) [reply tripy/CONTRIBUTING.md:122]: That's intentional. We build `mlir-tensorrt` with 10.2 and can link with 10.0+ (linking with 10.1 to keep it compatible with TRT-LLM).

File: tripy/CONTRIBUTING.md [hunk 2]
```diff
@@ -105,12 +105,43 @@ Please make sure any contributions you make satisfy the developer certificate of
 >		this project or the open source license(s) involved.
 
 
+### Use custom MLIR-TensorRT with Tripy
+Tripy depends on [MLIR-TensorRT](../mlir-tensorrt/README.md) for compilation and execution.
+Tripy container currently builds with [mlir-tensorrt-v0.1.29](https://github.com/NVIDIA/TensorRT-Incubator/releases/tag/mlir-tensorrt-v0.1.29), but you may
+choose to test Tripy with custom version of MLIR-TensorRT.
```
Comment (pranavm-nvidia) [inline tripy/CONTRIBUTING.md:111]:
```suggestion
The Tripy container currently builds with [mlir-tensorrt-v0.1.29](https://github.com/NVIDIA/TensorRT-Incubator/releases/tag/mlir-tensorrt-v0.1.29), but you may
choose to test Tripy with a custom version of MLIR-TensorRT.
```

File: tripy/CONTRIBUTING.md [hunk 3]
```diff
@@ -105,12 +105,43 @@ Please make sure any contributions you make satisfy the developer certificate of
...
+
+1. [Build custom MLIR-TensorRT](#contributing-to-mlir-tensorrt)
+
+2. Launch the container; from the [`tripy` root directory](.) with mlir-tensorrt repository mapped for accessing wheels files, run:
```
Comment (pranavm-nvidia) [inline tripy/CONTRIBUTING.md:115]:
```suggestion
2. Launch the container with mlir-tensorrt repository mapped for accessing wheels files; from the [`tripy` root directory](.), run:
```

File: tripy/CONTRIBUTING.md [hunk 4]
```diff
@@ -1,5 +1,16 @@
...
+
+## Table of Contents
+- [Setting Up](#setting-up)
+- [Making Changes](#making-changes)
+  - [Before You Start: Install pre-commit](#before-you-start-install-pre-commit)
+  - [Getting Up To Speed](#getting-up-to-speed)
+  - [Making Commits](#making-commits)
+- [Tests](#tests)
+- [Documentation](#documentation)
+- [Use Custom MLIR-TensorRT with Tripy](#use-custom-mlir-tensorrt-with-tripy)
```
Comment (pranavm-nvidia) [inline tripy/CONTRIBUTING.md:13]: GitHub autogenerates a table of contents. I think we can probably drop this?

**Review by parthchadha — APPROVED** [review]

File: tripy/CONTRIBUTING.md [hunk 5]
```diff
@@ -61,13 +62,22 @@ documentation.
...
-Ensure any commits you make are signed. See
-[this page](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#ssh-commit-signature-verification)
-for details on signing commits.
+1. Tripy follows [fork based developement](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo).
+    - Fork repo from GitHub
+    - Push changes to your branches on fork instead of the main repo
+    - Create a PR to merge changes from fork to the main repo
+
+2. Managing PRs
+    - Label your PR correctly (e.g., use `tripy` for `tripy` repository).
```
Comment (pranavm-nvidia) [inline tripy/CONTRIBUTING.md:74]:
```suggestion
    - Label your PR correctly (e.g., use `tripy` for changes to `tripy`).
```

**Review by pranavm-nvidia — APPROVED** [review]

**Pull request merged**